openprescribing/matrixstore/tests/test_grouper.py

```diff
@@ -0,0 +1,115 @@
+from itertools import product
+import random
+
+from django.test import SimpleTestCase
+
+from matrixstore.grouper import Grouper
+from matrixstore.matrix_ops import finalise_matrix, sparse_matrix
+
+
+class TestGrouper(SimpleTestCase):
```

Comment:
I wonder if this is a case where classic fixtures would be better than the compute-everything-with-a-different-code-path route?

The current approach has the advantage of programmatically covering all the edge cases, but the disadvantage that it's hard for a new reader to validate it's doing what we expect. This makes sense for things like measures because it's hard however you do it, but fixtures in this case are simple and easy to validate by reading, e.g.

```python
input_data = np.array([
    [1, 2, 3],
    [0, 1, 2],
    [0, 0, 0]]
)
groups = [(0, 'even'), (1, 'odd'), (2, 'even')]
expected = np.array([
    [1, 2, 3],
    [0, 1, 2]]
)
```

Reply:
I think the answer is just that we want both!